### PR TITLE
TA#15755 Add module hr_employee_type_private_wizard

### PR DESCRIPTION
**Pull request by ddufresne**

### CODE REVIEW

**Comment by ddufresne** [comment]
depends on https://github.com/Numigi/odoo-base-addons/pull/53

File: hr_employee_private_wizard/wizard/hr_employee_private_wizard.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        self._remove_non_private_arch_nodes(tree)
+        _make_name_field_readonly(tree)
+        _remove_image_field(tree)
+        tree.append(self._get_form_footer())
+        result["arch"] = etree.tostring(tree)
+        result["model"] = self._name
+        return result
+
+    def _get_form_footer(self):
+        footer = etree.fromstring(WIZARD_FOOTER)
```
Comment (foutoucour) [inline hr_employee_private_wizard/wizard/hr_employee_private_wizard.py:60]: Codacy Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/wizard/hr_employee_private_wizard.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+
+class EmployeePrivateWizard(models.TransientModel):
+
+    _name = "hr.employee.private.wizard"
+    _description = "Employee Private Data Wizard"
+
+    employee_id = fields.Many2one("hr.employee")
+
+    def save_and_close(self):
```
Comment (foutoucour) [inline hr_employee_private_wizard/wizard/hr_employee_private_wizard.py:22]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/tests/test_private_wizard.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# © 2020 - Today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from lxml import etree
```
Comment (foutoucour) [inline hr_employee_private_wizard/tests/test_private_wizard.py:4]: Codacy Issue found: [Using etree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace etree with the equivalent defusedxml package.](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/wizard/hr_employee_private_wizard.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
+# © 2020 - Today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from lxml import etree
```
Comment (foutoucour) [inline hr_employee_private_wizard/wizard/hr_employee_private_wizard.py:4]: Codacy Issue found: [Using etree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace etree with the equivalent defusedxml package.](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/wizard/hr_employee_private_wizard.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        tree = etree.fromstring(result["arch"])
+        self._remove_non_private_arch_nodes(tree)
+        _make_name_field_readonly(tree)
+        _remove_image_field(tree)
+        tree.append(self._get_form_footer())
+        result["arch"] = etree.tostring(tree)
+        result["model"] = self._name
+        return result
+
+    def _get_form_footer(self):
```
Comment (foutoucour) [inline hr_employee_private_wizard/wizard/hr_employee_private_wizard.py:59]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/tests/test_private_wizard.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    def test_private_field_in_fields_view_get_arch(self):
+        result = self.wizard.fields_view_get(view_type="form")
+        tree = result["arch"]
+        field = etree.fromstring(tree).xpath("//field[@name='gender']")[0]
+        assert field.attrib["modifiers"] == "{}"
+
+    def test_public_field_in_fields_view_get_arch(self):
+        result = self.wizard.fields_view_get(view_type="form")
+        tree = result["arch"]
+        field = etree.fromstring(tree).xpath("//field[@name='category_ids']")[0]
```
Comment (foutoucour) [inline hr_employee_private_wizard/tests/test_private_wizard.py:40]: Codacy Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_type_private_wizard/models/res_partner.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models, _
+from odoo.exceptions import AccessError
+from odoo.osv.expression import AND, OR
```
Comment (foutoucour) [inline hr_employee_type_private_wizard/models/res_partner.py:6]: Codacy Issue found: [Unused AND imported from odoo.osv.expression](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/tests/test_private_wizard.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        assert "gender" in result["fields"]
+
+    def test_private_field_in_fields_get(self):
+        fields = self.wizard.fields_get()
+        assert "gender" in fields
+
+    def test_private_field_in_fields_view_get_arch(self):
+        result = self.wizard.fields_view_get(view_type="form")
+        tree = result["arch"]
+        field = etree.fromstring(tree).xpath("//field[@name='gender']")[0]
```
Comment (foutoucour) [inline hr_employee_private_wizard/tests/test_private_wizard.py:34]: Codacy Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: hr_employee_private_wizard/wizard/hr_employee_private_wizard.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    def write(self, vals):
+        self._get_employee().write(vals)
+        return {"type": "ir.actions.act_window_close"}
+
+    @api.model
+    def fields_view_get(self, *args, **kwargs):
+        result = (
+            self.env["hr.employee"].sudo().fields_view_get(view_type="form")
+        )
+        tree = etree.fromstring(result["arch"])
```
Comment (foutoucour) [inline hr_employee_private_wizard/wizard/hr_employee_private_wizard.py:50]: Codacy Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5358386)

File: docker-compose.yml [hunk 10]
```diff
@@ -7,6 +7,12 @@ services:
     volumes:
       - odoo-web-data:/var/lib/odoo
       - ./.log:/var/log/odoo
+      - ../base-addons/base_extended_security:/mnt/extra-addons/base_extended_security
+      - ../base-addons/private_data_group:/mnt/extra-addons/private_data_group
+      - ../base-addons/test_http_request:/mnt/extra-addons/test_http_request
+      - ./hr_employee_private_wizard:/mnt/extra-addons/hr_employee_private_wizard
+      - ./hr_employee_type:/mnt/extra-addons/hr_employee_type
+      - ./hr_employee_type_private_wizard:/mnt/extra-addons/hr_employee_type_private_wizard
```
Comment (foutoucour) [inline docker-compose.yml]:
```suggestion
```

File: hr_employee_private_wizard/README.rst [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+This brings a limitation with sensitive HR data.
+A user can either view / edit the sensitive data of all employees or no employee at all.
+
+Module Design
+-------------
+This module adds a wizard.
+
+This wizard is created from an employee record.
+
+When the wizard is open, it shows the sensitive data of the employee.
```
Comment (foutoucour) [inline hr_employee_private_wizard/README.rst]:
```suggestion
When the wizard is opened, it shows the sensitive data of the employee.
```

File: hr_employee_private_wizard/i18n/fr.po [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+#~ msgstr "Interne"
+
+#~ msgid "Internal Employee"
+#~ msgstr "Employé interne"
+
+#~ msgid "Internal Employees"
+#~ msgstr "Employés internes"
+
+#~ msgid "Type"
+#~ msgstr "Type"
```
Comment (foutoucour) [inline hr_employee_private_wizard/i18n/fr.po]:
```suggestion
```

File: hr_employee_type/views/hr_employee.xml [hunk 13]
```diff
@@ -6,8 +6,10 @@
         <field name="inherit_id" ref="hr.view_employee_form"/>
         <field name="arch" type="xml">
             <field name="category_ids" position="after">
-                <group name="hr_type">
-                    <field name="ttype"/>
+                <group>
```
Comment (foutoucour) [inline hr_employee_type/views/hr_employee.xml:9]: pourquoi enlever le name du `group` ? 
ça me semblait simplifier l'utilisation des xpath non ?

Comment (ddufresne) [reply hr_employee_type/views/hr_employee.xml:9]: @foutoucour regarde une ligne plus bas. Je n'ai pas enlevé le name. J'ai imbriqué le groupe dans un autre groupe.

Comme ça, le champ prend seulement la moitié du formulaire en largeur.

File: hr_employee_private_wizard/README.rst [hunk 14]
```diff
@@ -17,7 +17,7 @@ This module adds a wizard.
 
 This wizard is created from an employee record.
 
-When the wizard is open, it shows the sensitive data of the employee.
+When the wizard is <span class="x x-first x-last">opened</span>, it shows the sensitive data of the employee.
```
Comment (foutoucour) [inline hr_employee_private_wizard/README.rst:20]:
```suggestion
When the wizard is opened, it shows the sensitive data of the employee.
```
la suggestion de github a fait un peu n'importe quoi ici, dit donc.

**Review by foutoucour — APPROVED** [review]

**Pull request merged**